data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApiWithSwaggerTest.java

```diff
@@ -28,12 +28,13 @@
 import tech.pegasys.artemis.api.DataProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 import tech.pegasys.artemis.storage.CombinedChainDataClient;
+import tech.pegasys.artemis.storage.api.DiskUpdateChannel;
 import tech.pegasys.artemis.sync.SyncService;
 import tech.pegasys.artemis.util.config.ArtemisConfiguration;
 
 public class BeaconRestApiWithSwaggerTest {
   private final ChainStorageClient storageClient =
-      ChainStorageClient.memoryOnlyClient(new EventBus());
+      ChainStorageClient.memoryOnlyClient(new EventBus(), mock(DiskUpdateChannel.class));
```

Comment:
I wonder if we should create a text fixture class that creates a memory only `ChainStorageClient` and then it can automatically create the mock `DiskUpdateChannel`.  We never use the memory only version in production so it's weird that we've polluted our production code with test setup constructors.  Might be a separate PR though.

Reply:
I highly agree. I actually want to do it in this PR. ChainStorageClient would be way more legible if cleaned from the non-production code.

Reply:
Actually, I will merge this in and quickly do a follow-up PR. I have a feeling that, that PR might get a lot of comments 😂